cmake/AutoOptionHelpers.cmake

```diff
@@ -128,11 +140,17 @@ macro(auto_option _FEATURE_NAME)
     message(FATAL_ERROR "Invalid auto_option default ${ARG_DEFAULT}")
   endif()
 
+  if(ARG_COMPONENT)
+    set(COMPONENT ${ARG_COMPONENT})
+  else()
+    set(COMPONENT OFF)
```

Comment:
I'm not sure about setting it to `OFF` here; it doesn't represent an option.

Reply:
If I don't set it to off, the condition picking the COMPONENTS path would not work